Source/WebCore/inspector/agents/InspectorPageAgent.cpp

```diff
@@ -753,7 +765,7 @@ Inspector::Protocol::ErrorStringOr<void> InspectorPageAgent::setCookie(Ref<JSON:
         auto* page = document->page();
         if (!page)
             continue;
-        page->cookieJar().setRawCookie(*document, cookie.value());
+        page->cookieJar().setRawCookie(*document, cookie, isPartitioned);
```

Comment:
wait actually is there really a need to include `partitioned` in `Page.Cookie` if it's only ever used by `Page.setCookie`?

what if instead we had `partitioned` just be another parameter for `Page.setCookie` since ultimately that's really only where it's needed (because in all other places we can assume `partitioned` if `partitionKey` is not empty)
```cpp
Inspector::Protocol::ErrorStringOr<void> InspectorPageAgent::setCookie(Ref<JSON::Object>&& cookieObject, std::optional<bool>&& shouldPartition)
{
    /* ... */
    auto isPartitioned = (!cookie->partitionKey.isEmpty() || shouldPartition.value_or(false)) ? IsPartitionedCookie::Yes : IsPartitionedCookie::No;
    /* ... */
        page->cookieJar().setRawCookie(*document, cookie.value(), isPartitioned);
    /* ... */
}
```
(and to be clear with this approach we'd only add `partitionKey` to `Page.Cookie`)

Reply:
Good point. That does simplify this a bit.